lib/SPIRV/OCLTypeToSPIRV.cpp

```diff
@@ -250,8 +250,12 @@ void OCLTypeToSPIRV::adaptArgumentsBySamplerUse(Module &M) {
           AdaptedTy.count(SamplerArg) != 0) // Already traced this, move on.
         continue;
 
+      if (isSPIRVType(SamplerArg->getType(), kSPIRVTypeName::Sampler))
+        return;
```

Comment:
We only need to adapt the i32 type sampler argument in __spirv_SampledImage. This only happens with the old SPIR 1.2/2.0 format.

If the sampler arg is already spirv.Sampler type, that means FE already generates correct type, so we do not need to adapt it.